src/main/java/uk/gov/companieshouse/api/accounts/service/impl/DebtorsService.java

```diff
@@ -38,23 +40,40 @@
     private DebtorsTransformer transformer;
     private SmallFullService smallFullService;
     private KeyIdGenerator keyIdGenerator;
+    private DebtorsValidator debtorsValidator;
 
     @Autowired
     public DebtorsService(DebtorsRepository repository,
                           DebtorsTransformer transformer,
                           SmallFullService smallFullService,
-                          KeyIdGenerator keyIdGenerator) {
+                          KeyIdGenerator keyIdGenerator, DebtorsValidator debtorsValidator) {
 
         this.repository = repository;
         this.transformer = transformer;
         this.smallFullService = smallFullService;
         this.keyIdGenerator = keyIdGenerator;
+        this.debtorsValidator = debtorsValidator;
     }
 
     @Override
     public ResponseObject<Debtors> create(Debtors rest, Transaction transaction,
                                           String companyAccountsId, HttpServletRequest request)
-        throws DataException {
+        throws DataException, RestException {
+
+        Errors errors;
+
+        try {
+            errors = debtorsValidator.validateDebtors(rest, transaction);
```

Comment:
Is there any specific reason we need a RestException at this point ? I am just comparing it to the other Validators in the code, not sure if this is needed unless there is a strong reason for it.

Reply:
We use the custom exception for any errors from retrieving the company profile data from the company profile api in the validator. So far this is the first note that needs to do this as it has a previous period - which we need to check is valid by checking if the company is a multiple year filer. I believe future notes will follow suit.